src/api/routes/user.ts

```diff
@@ -0,0 +1,97 @@
+import "reflect-metadata";
+import express, { Router, Request, Response, NextFunction } from "express";
+import Container from "typedi";
+import { UserValidatorService } from "../validators/user.validator";
+
+import { UserService } from "../../services/user.service";
+import { IdGeneratorService } from "../../services/id-generator.service";
+
+export class UserRoutes {
+  private router = Router();
+
+  private userService = Container.get(UserService);
+  private generateId = Container.get(IdGeneratorService).generateId;
+  private validator = Container.get(UserValidatorService);
+
+  constructor(private app: Router) {
+    this.init();
+  }
+
+  private init(): void {
+    this.app.use("/user", this.router);
+
+    this.router.post(
+      "/",
+      this.generateId,
+      this.validator.checkCreateUser,
+      (req: Request, res: Response) => {
+        this.userService.createUser(req.body);
+        res
+          .json({ message: `User ${req.body.login} was created successfully.` })
+          .status(200);
+      }
+    );
+
+    this.router.get(
+      "/list",
+      this.validator.checkGetUserList,
+      (req: Request, res: Response, next: NextFunction) => {
+        const usersList = this.userService.getSortedUserList(req.headers);
+        if (usersList.length) {
+          res.json({ userList: usersList }).status(200);
+          return;
```

Comment:
It is good practice to return last operation in one-line

Reply:
Done.